Library/ViewModels/DiscoveryViewModel.swift

```diff
@@ -64,23 +73,23 @@ public final class DiscoveryViewModel: DiscoveryViewModelType, DiscoveryViewMode
 DiscoveryViewModelOutputs {
   fileprivate static let defaultParams = .defaults |> DiscoveryParams.lens.includePOTD .~ true
 
-  // swiftlint:disable function_body_length
+  // swiftlint:disable:next function_body_length
   public init() {
     let sorts: [DiscoveryParams.Sort] = [.magic, .popular, .newest, .endingSoon, .mostFunded]
 
     self.configurePagerDataSource = self.viewDidLoadProperty.signal.mapConst(sorts)
     self.configureSortPager = self.configurePagerDataSource
 
-    let currentParams = self.viewWillAppearProperty.signal
-      .take(first: 1)
-      .flatMap { [filterWithParams = filterWithParamsProperty.producer.skipNil()] _ in
-        filterWithParams.prefix(value: DiscoveryViewModel.defaultParams)
-      }
+    let currentParams = Signal.merge(
+      self.viewWillAppearProperty.signal.take(first: 1).mapConst(DiscoveryViewModel.defaultParams),
+      self.filterWithParamsProperty.signal.skipNil()
+      )
```

Comment:
just a small refactor while i was here. don't know why i jumped through hoops for that weird producer thing when it can be done so much more easily...